Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -0,0 +1,759 @@
+define([
+        '../Core/ApproximateTerrainHeights',
+        '../Core/Check',
+        '../Core/ComponentDatatype',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/DeveloperError',
+        '../Core/GeometryInstance',
+        '../Core/GeometryInstanceAttribute',
+        '../Core/GroundPolylineGeometry',
+        '../Core/isArray',
+        '../Core/Matrix4',
+        '../Shaders/PolylineShadowVolumeVS',
+        '../Shaders/PolylineShadowVolumeFS',
+        '../Shaders/PolylineShadowVolumeMorphVS',
+        '../Shaders/PolylineShadowVolumeMorphFS',
+        '../Renderer/DrawCommand',
+        '../Renderer/Pass',
+        '../Renderer/RenderState',
+        '../Renderer/ShaderProgram',
+        '../Renderer/ShaderSource',
+        '../ThirdParty/when',
+        './BlendingState',
+        './CullFace',
+        './Material',
+        './PolylineColorAppearance',
+        './PolylineMaterialAppearance',
+        './Primitive',
+        './SceneMode'
+    ], function(
+        ApproximateTerrainHeights,
+        Check,
+        ComponentDatatype,
+        defaultValue,
+        defined,
+        defineProperties,
+        destroyObject,
+        DeveloperError,
+        GeometryInstance,
+        GeometryInstanceAttribute,
+        GroundPolylineGeometry,
+        isArray,
+        Matrix4,
+        PolylineShadowVolumeVS,
+        PolylineShadowVolumeFS,
+        PolylineShadowVolumeMorphVS,
+        PolylineShadowVolumeMorphFS,
+        DrawCommand,
+        Pass,
+        RenderState,
+        ShaderProgram,
+        ShaderSource,
+        when,
+        BlendingState,
+        CullFace,
+        Material,
+        PolylineColorAppearance,
+        PolylineMaterialAppearance,
+        Primitive,
+        SceneMode) {
+    'use strict';
+
+    /**
+     * A GroundPolylinePrimitive represents a polyline draped over the terrain in the {@link Scene}.
+     * <p>
+     *
+     * Only to be used with GeometryInstances containing GroundPolylineGeometries
+     *
+     * @param {Object} [options] Object with the following properties:
+     * @param {Array|GeometryInstance} [options.geometryInstances] GeometryInstances containing GroundPolylineGeometry
+     * @param {Appearance} [options.appearance] The Appearance used to render the polyline. Defaults to a white color {@link Material} on a {@link PolylineMaterialAppearance}.
+     * @param {Boolean} [options.show=true] Determines if this primitive will be shown.
+     * @param {Boolean} [options.vertexCacheOptimize=false] When <code>true</code>, geometry vertices are optimized for the pre and post-vertex-shader caches.
```

Comment:
I wonder if this is needed.  Are the indices already in good shape based on how you generated them and that this is a polyline?

Reply:
It's still possible that a single primitive could have a lot of triangles, but I guess tipsify isn't as relevant here since the geometry in a given view should be pretty sparse and since early Z is disabled. I'll take this out.